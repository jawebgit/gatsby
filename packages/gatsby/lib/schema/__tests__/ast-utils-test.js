const { extractFieldExamples, buildFieldEnumValues } = require(`../ast-utils`)

describe(`Gatsby AST utils`, () => {
  const nodes = [
    {
      name: `The Mad Max`,
      hair: 1,
      date: `2006-07-22T22:39:53.000Z`,
      anArray: [1, 2, 3, 4],
      frontmatter: {
        date: `2006-07-22T22:39:53.000Z`,
        title: `The world of dash and adventure`,
        blue: 100,
      },
    },
    {
      name: `The Mad Wax`,
      hair: 2,
      date: `2006-07-22T22:39:53.000Z`,
      anArray: [1, 2, 5, 4],
      frontmatter: {
        date: `2006-07-22T22:39:53.000Z`,
        title: `The world of slash and adventure`,
        blue: 10010,
      },
    },
  ]

  it(`builds field examples from an array of nodes`, () => {
    expect(extractFieldExamples({ nodes })).toMatchSnapshot()
  })

  it(`build enum values for fields from array on nodes`, () => {
    expect(buildFieldEnumValues(nodes)).toMatchSnapshot()
  })
})
