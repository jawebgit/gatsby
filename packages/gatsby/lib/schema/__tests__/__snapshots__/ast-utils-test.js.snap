exports[`Gatsby AST utils build enum values for fields from array on nodes 1`] = `
Object {
  "anArray": Object {
    "field": "anArray",
  },
  "date": Object {
    "field": "date",
  },
  "frontmatter___blue": Object {
    "field": "frontmatter.blue",
  },
  "frontmatter___date": Object {
    "field": "frontmatter.date",
  },
  "frontmatter___title": Object {
    "field": "frontmatter.title",
  },
  "hair": Object {
    "field": "hair",
  },
  "name": Object {
    "field": "name",
  },
}
`;

exports[`Gatsby AST utils builds field examples from an array of nodes 1`] = `
Object {
  "anArray": Array [
    1,
    2,
    3,
    4,
  ],
  "date": "2006-07-22T22:39:53.000Z",
  "frontmatter": Object {
    "blue": 100,
    "date": "2006-07-22T22:39:53.000Z",
    "title": "The world of dash and adventure",
  },
  "hair": 1,
  "name": "The Mad Max",
}
`;
